addons/events/fnc_addPlayerEventHandler.sqf

```diff
@@ -121,115 +140,134 @@ if (_id != -1) then {
         GVAR(oldGroup) = grpNull;
         GVAR(oldLeader) = objNull;
         GVAR(oldWeapon) = "";
+        GVAR(oldMuzzle) = [objNull, "", ""];
+        GVAR(oldWeaponMode) = [objNull, GVAR(oldMuzzle), ""];
         GVAR(oldLoadout) = [];
         GVAR(oldLoadoutNoAmmo) = [];
         GVAR(oldVehicle) = objNull;
-        GVAR(inVehicle) = false;
         GVAR(oldTurret) = [];
         GVAR(oldVisionMode) = -1;
         GVAR(oldCameraView) = "";
         GVAR(oldFeatureCamera) = "";
         GVAR(oldVisibleMap) = false;
-        GVAR(oldUAVControl) = [];
-        GVAR(controlledEntity) = objNull;
+        GVAR(oldState) = [];
 
         GVAR(playerEHInfo) pushBack addMissionEventHandler ["EachFrame", {call FUNC(playerEH_EachFrame)}];
         [QFUNC(playerEH_EachFrame), {
             SCRIPT(playerEH_EachFrame);
-            private _player = missionNamespace getVariable ["bis_fnc_moduleRemoteControl_unit", player];
-            if !(_player isEqualTo GVAR(oldUnit)) then {
-                [QGVAR(unitEvent), [_player, GVAR(oldUnit)]] call CBA_fnc_localEvent;
-                GVAR(oldUnit) = _player;
-                GVAR(oldUAVControl) = []; // force update
-            };
-
-            private _data = group _player;
-            if !(_data isEqualTo GVAR(oldGroup)) then {
-                [QGVAR(groupEvent), [_player, GVAR(oldGroup)]] call CBA_fnc_localEvent;
-                GVAR(oldGroup) = _data;
-            };
+            private _unit = missionNamespace getVariable ["bis_fnc_moduleRemoteControl_unit", player];
+            private _vehicle = vehicle _unit;
+
+            private _controlledEntity = _unit;
+            if (!isNull getConnectedUAV _unit) then {
+                private _uavControl = UAVControl getConnectedUAV _unit;
+                _uavControl = _uavControl param [(_uavControl find _unit) + 1, ""];
```

Comment:
Maybe add a little warning, it works fine because of `isEqualTo`
```suggestion
                _uavControl = _uavControl param [(_uavControl find _unit) + 1, ""]; // will be position string if actively controlling, or object if not
```

Reply:
Right, since -1 + 1 = 0, it reports the first unit instead of the players role.